### PR TITLE
Fixes cigarette pipes spawning infinte ashes and no more lighter spam

### LINKED ISSUE #9578: Smoking pipes don't empty properly
image

I would assume the logical outcome is that after you empty the ash, the pipe's contents are also emptied, but this is not the case - you can relight the burnt out goods and shortly after they burn out again, producing more and more ash on the floor.

### PR DESCRIPTION
**Pull request by alsoandanswer**
Fixes #9578

Lighter on/off now is on a timer. So your logs won't get filled up.

### CODE REVIEW

**Comment by alsoandanswer** [comment]
!bugfix

**Comment by alsoandanswer** [comment]
!review

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 1]
```diff
@@ -253,12 +253,12 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 			if(!no_message)
 				to_chat(M, SPAN_NOTICE("Your [name] goes out, and you empty the ash."))
 				playsound(src.loc, 'sound/items/cigs_lighters/cig_snuff.ogg', 50, 1)
-			lit = 0
-			icon_state = icon_off
-			item_state = icon_off
 			M.update_inv_wear_mask(0)
 			M.update_inv_l_hand(0)
 			M.update_inv_r_hand(1)
+		lit = 0
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/cigs_lighters.dm:259]:
```suggestion
		lit = FALSE
```

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 2]
```diff
@@ -691,31 +693,38 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 			lit = TRUE
 			update_icon()
 			playsound(src.loc, pick(activation_sound), 75, 1)
-			if(istype(src, /obj/item/flame/lighter/zippo) )
-				user.visible_message(FONT_SMALL("Without even breaking stride, <b>[user]</b> flips open and lights \the [src] in one smooth movement."), range = 3)
-			else
+			if(istype(src, /obj/item/flame/lighter/zippo))
+				if(last_open <= world.time - 20) //Spam limiter.
+					last_open = world.time
+					user.visible_message(SPAN_NOTICE("Without even breaking stride, <b>[user]</b> flips open and lights \the [src] in one smooth movement."), range = 3)
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/cigs_lighters.dm:699]: I can't say I like how you reverted these, the previous ones looked fine, but oh well

Comment (alsoandanswer) [reply code/game/objects/items/weapons/cigs_lighters.dm:699]: The burn message felt kinda out of place, if you want a reason.

**Pull request merged**